crypto/s2n_hmac.c

```diff
@@ -192,6 +178,27 @@ int s2n_hmac_init(struct s2n_hmac_state *state, s2n_hmac_algorithm alg, const vo
     return s2n_hmac_reset(state);
 }
 
+int s2n_hmac_init(struct s2n_hmac_state *state, s2n_hmac_algorithm alg, const void *key, uint32_t klen)
+{
+    state->alg = alg;
+    GUARD(s2n_hmac_block_size(alg, &state->block_size));
+    state->currently_in_hash_block = 0;
+    GUARD(s2n_hmac_hash_block_size(alg, &state->hash_block_size));
+    GUARD(s2n_hmac_digest_size(alg, &state->digest_size));
+
+    gte_check(sizeof(state->xor_pad), state->block_size);
+    gte_check(sizeof(state->digest_pad), state->digest_size);
+    /* key needs to be as large as the biggest block size */
+    gte_check(sizeof(state->xor_pad), state->hash_block_size);
```

Comment:
I discussed this with @alexw91, and we agree that this new check is not needed and is a no-op. For HMAC, `block_size` (as per [PR 570](https://github.com/awslabs/s2n/pull/570), actually `xor_pad_size`) will always equal `hash_block_size`. For SSLv3 MAC, `block_size` is smaller, but the `xor_pad` is always written to and read from based on `block_size`.

Reply:
There is a code comment in the header about this, and I generally tend to find asserts more long term stable than comments, but can remove if you'd prefer.

Reply:
As discussed offline, this check can be added now given the current implementation. When Issue #572 is added, it should be safe to remove this check.